zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/utils/ObjectSizeEstimator.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.utils;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Output;
+import io.camunda.webapps.schema.entities.ExporterEntity;
+import org.apache.commons.io.output.CountingOutputStream;
+import org.apache.commons.io.output.NullOutputStream;
+
+public class ObjectSizeEstimator {
+
+  private static final ThreadLocal<Kryo> THREAD_LOCAL_KRYO =
+      ThreadLocal.withInitial(
+          () -> {
+            final Kryo kryo = new Kryo();
+            kryo.setRegistrationRequired(false);
+            kryo.setReferences(false);
+            return kryo;
+          });
+
+  public static long kryoEstimate(final ExporterEntity<?> entity) {
```

Comment:
🔧 The signature shouldn't be bound to the underlying implementation, and in this case the library we are using. Something like `estimateSize` should be sufficient